server/claim/claim-revision/claim-revision.service.ts

```diff
@@ -204,17 +204,17 @@ export class ClaimRevisionService {
         if (sources && Array.isArray(sources)) {
             for (let source of sources) {
                 try {
-                    if (typeof source === "string") {
+                    const existingSource = await this.sourceService.findHref(source);
+                    if (existingSource && existingSource.length > 0) {
+                        for (let source of existingSource) {
```

Comment:
question: Would it be possible to optimize the loop execution by using Promise.all? Processing the updates concurrently instead of sequentially could potentially improve performance